src/main/java/com/gaggle/contact/controllers/ContactController.java

```diff
@@ -44,6 +46,20 @@
                 .orElseThrow(() -> new ContactNotFoundException(name));
     }
 
+    @GetMapping("/query")
+    public List<Contact> getByEmail(@RequestParam final String email) {
+        if (Objects.isNull(email) || !email.contains("@")) {
+            throw new RuntimeException("email is no good, try again silly");
+        }
+
+        Optional<List<Contact>> contacts = repository.findByEmailContaining(email);
+        if (contacts.get().isEmpty()) {
+            throw new RuntimeException(String.format("no contacts with email [%s] found.", email));
+        }
+
+        return contacts.get();
```

Comment:
Per requirement 4, first name and last name will have to be derived from name. Let's have a conversation about updating the table and refractoring code to reflect that, or have some code that will split that here. Preferably the former.